### ISSUE TITLE
"File drop only" for internally shared folders

### ISSUE THREAD

**Issue by FloMiLe** [nextcloud/server#16515]
**Is your feature request related to a problem? Please describe.**
I am using Nextcloud to share documents with my high school students, as well as to collect homework. For collection, the students are supposed to be able to drop files into a folder (for a given assignment), but should not see other students' documents. Therefore, I use a share link configured as file drop only. 

It frustrates me that I have to it this way, namely, to create a folder, configure it and distribute the link.

**Describe the solution you'd like**
I wish I could simply share a folder with a group and set it to file drop, i.e. the users in the group can see the folder and upload files to it, but cannot see its content.

**Describe alternatives you've considered**
The only other way to create a pure drop box would be as described above.

**Comment by A-Kasper** [comment]
- Everybody should see his own uploads. Than they would have validation, that the solution is uploaded correctly.

**Comment by melikyuksel** [comment]
Yes, it makes no sense for the file drop option to only be for externally shared links and not for sharing between internal users. If anything, it’d have been the other way around.

**Comment by szaimen** [comment]
cc @nextcloud/designers is this something that we should do and how could it work?

**Comment by melyux** [comment]
Not stale yet

**Comment by BDevGW** [comment]
Similar situation here.
We want a single folder where we can collect files from multiple workers but without them seeing each others files.
Solution is a link share that allows file drop but as this is a public link anyone can upload (and so spam) the folder if someone leaks that link.
Solution would be an option to allow only authenticated users to drop files.
Basically dedicated "can see the folder" and "can see the folders contents" permissions so I can say "can see and write to folder" without also saying "can see contents of folder".

**Comment by FloMiLe** [comment]
I don't think it's stale. I think the developers should make a conscious decision about whether implementing this is worthwhile...

**Comment by szaimen** [comment]
cc @jancborchardt @nimisha-vijay

**Comment by nimisha-vijay** [comment]
Do you mean similar permissions for internal user and group shares as there are for share links (read only, upload and editing, file drop)?

**Comment by BDevGW** [comment]
Yes. Filedrop for chosen authenticated users (or groups) only.

**Comment by tobiasKaminsky** [comment]
We currently have no idea how to do this on Desktop.
Imagine you receive such a shared "file drop only" folder and put a file into it, but then 
- you cannot see it after next refresh (same on web ui)
- you cannot remove it
- you do not know that this is such a "file drop only" folder

Do you have an idea how to inform user? As far as I know there is no way to show such an info on file explorers on Linux, Windows or MacOS.

**Comment by melyux** [comment]
This stale bot is overly zealous.

**Comment by bes1002t** [comment]
@tobiasKaminsky the Desktop client don't have to know that. The client is logged into the backend and the backend just have to provide all files the user uploaded instead of the whole directory content.

> "you cannot see it after next refresh (same on web ui)"
why? Does the Explorer and Webui not know about which user uploaded a file?

**Comment by melyux** [comment]
Mm no

**Comment by melyux** [comment]
No

**Comment by melyux** [comment]
No

**Comment by jancborchardt** [comment]
@tobiasKaminsky what do you think about the remarks from @melyux and @bes1002t?

**Comment by tobiasKaminsky** [comment]
> @tobiasKaminsky what do you think about the remarks from @melyux and @bes1002t?

This helps a bit, I suppose, but as you can see, people need to ask (otherwise there would have been no SO question in first place).



> One problem that definitely would still stand is "you do not know that this is such a 'file drop only' folder".

Maybe we can auto-create a read-only Readme.md which is always shown and there we inform users about this concept.

Btw: this is also not that clear for users in web:
![image](https://user-images.githubusercontent.com/5836855/152297630-a1056428-df40-4682-a617-0b7f563e229c.png) ![image](https://user-images.githubusercontent.com/5836855/152297678-a1d46d10-74f5-46aa-84a6-1ed94f67acc5.png)
